edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet.Docker/SystemResourcesMetrics.cs

```diff
@@ -179,12 +179,19 @@ void SetModuleStats(SystemResources systemResources)
                 });
                 module.MemoryStats.ForEach(ms =>
                 {
-                    ms.Limit.ForEach(limit => this.totalMemory.Set(limit, tags));
-                    ms.Usage.ForEach(usage =>
+                    if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
```

Comment:
should we consider to separate in different classes for different platforms? do we need to change this code to support Yocto?

Reply:
It's likely docker would have a different json output for something like Yocto, so for now I'm just worried about tier 1

Reply:
do we enable/test metrics in either new or old e2e tests?

Reply:
Yep. There's tests in the new e2e framework, which I'm currently expanding